models/project_issues.go

```diff
@@ -0,0 +1,206 @@
+func (p *Project) NumClosedIssues() int {
+	c, err := x.Table("project_issues").
+		Join("INNER", "issue", "project_issues.issue_id=issue.id").
+		Where("project_issues.project_id=? AND issue.is_closed=?", p.ID, true).Count("issue.id")
+	if err != nil {
+		return 0
+	}
+	return int(c)
+}
+
+// NumOpenIssues return counter of open issues assigned to a project
+func (p *Project) NumOpenIssues() int {
+	c, err := x.Table("project_issues").
+		Join("INNER", "issue", "project_issues.issue_id=issue.id").
+		Where("project_issues.project_id=? AND issue.is_closed=?", p.ID, false).Count("issue.id")
+	if err != nil {
+		return 0
+	}
+	return int(c)
+}
+
+// ChangeProjectAssign changes the project associated with an issue
+func ChangeProjectAssign(issue *Issue, doer *User, newProjectID int64) error {
+
+	sess := x.NewSession()
+	defer sess.Close()
+	if err := sess.Begin(); err != nil {
+		return err
+	}
+
+	if err := addUpdateIssueProject(sess, issue, doer, newProjectID); err != nil {
+		return err
+	}
+
+	return sess.Commit()
+}
+
+func addUpdateIssueProject(e *xorm.Session, issue *Issue, doer *User, newProjectID int64) error {
+
+	oldProjectID := issue.projectID(e)
+
+	if _, err := e.Where("project_issues.issue_id=?", issue.ID).Delete(&ProjectIssues{}); err != nil {
+		return err
+	}
+
+	if err := issue.loadRepo(e); err != nil {
+		return err
+	}
+
+	if oldProjectID > 0 || newProjectID > 0 {
+		if _, err := createComment(e, &CreateCommentOptions{
+			Type:         CommentTypeProject,
+			Doer:         doer,
+			Repo:         issue.Repo,
+			Issue:        issue,
+			OldProjectID: oldProjectID,
+			ProjectID:    newProjectID,
+		}); err != nil {
+			return err
+		}
+	}
+
+	_, err := e.Insert(&ProjectIssues{
+		IssueID:   issue.ID,
+		ProjectID: newProjectID,
+	})
+	return err
+}
+
+//  ____            _           _   ____                      _
+// |  _ \ _ __ ___ (_) ___  ___| |_| __ )  ___   __ _ _ __ __| |
+// | |_) | '__/ _ \| |/ _ \/ __| __|  _ \ / _ \ / _` | '__/ _` |
+// |  __/| | | (_) | |  __/ (__| |_| |_) | (_) | (_| | | | (_| |
+// |_|   |_|  \___// |\___|\___|\__|____/ \___/ \__,_|_|  \__,_|
+//               |__/
+
+// MoveIssueAcrossProjectBoards move a card from one board to another
+func MoveIssueAcrossProjectBoards(issue *Issue, board *ProjectBoard) error {
+
+	sess := x.NewSession()
+	defer sess.Close()
+	if err := sess.Begin(); err != nil {
+		return err
+	}
+
+	var pis ProjectIssues
+	has, err := sess.Where("issue_id=?", issue.ID).Get(&pis)
+	if err != nil {
+		return err
+	}
+
+	if !has {
+		return fmt.Errorf("issue has to be added to a project first")
+	}
+
+	pis.ProjectBoardID = board.ID
+	if _, err := sess.ID(pis.ID).Cols("project_board_id").Update(&pis); err != nil {
+		return err
+	}
+
+	return sess.Commit()
+}
+
+func (pb *ProjectBoard) removeIssues(e Engine) error {
+	_, err := e.Exec("UPDATE `project_issues` SET project_board_id = 0 WHERE project_board_id = ? ", pb.ID)
```

Comment:
As mentioned above - I suspect that the project_board_id index won't be a good index and we should instead rely on subsearching through the projectID's boards